loss_plots.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+import os
+import glob
+import sys
+import matplotlib.pyplot as plt
+from mpl_toolkits import mplot3d
+
+
+plt.rcParams['figure.figsize'] = (12, 9)
+plt.rcParams['figure.edgecolor'] = 'k'
+plt.rcParams['figure.facecolor'] = 'w'
+plt.rcParams['savefig.dpi'] = 400
+plt.rcParams['savefig.bbox'] = 'tight'
+plt.rcParams['text.usetex'] = True
+plt.rcParams['font.family'] = "serif"
+
+
+# Optimization Sets
+radius_set = [0.5, 0.7, 0.9, 1, 1.1, 1.2, 1.3, 1.5]
+noise_set = [0.0001, 0.0003, 0.0007, 0.001, 0.003, 0.005, 0.007, 0.01]
+
+# radius_set = [0.1, 0.5, 1]
+# noise_set = [0.001, 0.0007, 0.003]
+
+reservoir_set = [600, 800, 1000, 1500, 2000, 2500, 3000, 4000]
+sparsity_set = [0.005, 0.01, 0.03, 0.05, 0.1, 0.12, 0.15, 0.2]
+
+# reservoir_set = [600, 800, 1000]
+# sparsity_set = [0.005, 0.01, 0.2]
+
+parameter_sets = {"reservoir": reservoir_set,
+                  "sparsity": sparsity_set,
+                  "rho": radius_set,
+                  "noise": noise_set}
+
+variables = {'reservoir': 'Reservoir Size',
+             'sparsity': 'Sparsity',
+             'rho': 'Spectral Radius',
+             'noise': 'Noise',
+             'trainlen': 'Training Length'}
+
+
+def get_loss_data():
+    """
+    This function returns a list of paths to all .npy loss
+    files.
+
+    Returns
+    -------
+    path_list : list of strings
+        The list of paths to output files
+    """
+
+    path = "./data/*_loss.npy"
+
+    path_list = glob.glob(path, recursive=True)
+    return path_list
+
+
+def get_variable_sets(fname):
+    """
+    This function returns the variable set given a
+    file name of loss data.
+
+    Parameters
+    ----------
+    fname : string
+        The file name of the data. This string is
+        parsed and should include the model parameters being
+        optimized.
+
+    Returns
+    -------
+    xset : list
+        The list of values for the first variable
+    xvar : string
+        The name of the first variable
+    yset : list
+        The list of values for the second variable
+    yvar : string
+        The name of the second variable
+    figure_path : string
+    """
+    splitstring = fname.split('_')
+
+    xset, yset = None, None
+    xvar, yvar = None, None
+
+    # print(fname)
```

Comment:
It feels like you used this for debugging the script.
If the script has the desirable behavior, you should remove the line.

Reply:
The same comment applies to other commented prints.